tests/integration_tests.rs

```diff
@@ -900,6 +900,60 @@ fn simple_for_dictmerger_loop() {
     unsafe { weld_value_free(ret_value) };
 }
 
+fn simple_for_global_dictmerger_loop() {
```

Comment:
Is this function just a copy of `simple_for_dictmerger_loop` with the `global` annotation? If so, we should try to share the code between the two tests.